### PR TITLE
dotnet: check formatting on CI

### PR DESCRIPTION
**Pull request by matklad**
Also remove BOM and translate CRLF -> LF

### CODE REVIEW

**Comment by matklad** [comment]
@batiati do you know by any chance to force dotnet format use \n without explicit config through .editorconfig?

**Comment by batiati** [comment]
> @batiati do you know by any chance to force dotnet format use \n without explicit config through .editorconfig?

I'm afraid we need a .editorconfig file to parameterize it. The bright side is that we now can have a consistent format across Linux and Windows, and VSCode or Visual Studio as well.
Sometimes I use Visual Studio on Windows, and it's so frustrating when it creates new files with CRLF despite all other files being LF.

File: src/clients/dotnet/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# By default, `dotnet format` uses `\r\n` line endings on Windows. To override this behavior, we
+# need this editorconfig file.
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
```
Comment (batiati) [inline src/clients/dotnet/.editorconfig:7]: Let's be more opinionated:

```suggestion
end_of_line = lf
insert_final_newline = true
charset=utf-8
indent_style=space
indent_size=4
```

Reference:

- [charset](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#charset)
- [indent_style](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#indent_style)
- [indent_size](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#indent_size)

Comment (matklad) [reply src/clients/dotnet/.editorconfig:7]: Originally I thought to stick to defaults as much as possible, but, indeed, if we have to configure something, we could as well pin down more things!

**Review by batiati — APPROVED** [review]

**Pull request merged**